modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java

```diff
@@ -336,18 +407,22 @@ private Connection getConnection(String user, String password, String url) throw
     public void dropDatabase(Map<String, Object> params) throws IOException {
         JDBCDataStore closer = new JDBCDataStore();
         // get the connection params
-        String host = (String) HOST.lookUp(params);
-        int port = (Integer) PORT.lookUp(params);
         String db = (String) DATABASE.lookUp(params);
         String user = (String) USER.lookUp(params);
         String password = (String) PASSWD.lookUp(params);
+        // TODO make a copy of the map (hash map ?)
```

Comment:
Please don't modify the values provided by user, the database.key should not be touched.

Reply:
The idea here, is to copy the user input, in order to change the db from something to template1. I would never change the initial value of the object, just making a local copy of it, in order to pass this specific case.